src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -56,76 +57,34 @@ class HttpGoogleCloudStorageDAO(
-    val serviceCredential = getGroupServiceAccountCredential
-    val directory = getGroupDirectory(serviceCredential)
-    directory.groups.delete(groupId).execute()
-  }
-
-  override def createBucket(userInfo: UserInfo, projectId: String, bucketName: String): Unit = {
-    val credential = getBucketCredential(userInfo)
-    val storage = getStorage(credential)
-    val bucket =  new Bucket().setName(bucketName)
-    storage.buckets().insert(projectId, bucket).execute()
+  override def createBucket(userInfo: UserInfo, projectId: String): Try[String] = {
+    val buckets = getStorage(getBucketCredential(userInfo)).buckets
+    retry(2) { // in case of bucket name collisions
+      val bucketName = createBucketName
+      buckets.insert(projectId, new Bucket().setName(bucketName)).execute
+      bucketName
+    }
   }
 
-  override def deleteBucket(userInfo: UserInfo, projectId: String, bucketName: String): Unit = {
+  override def deleteBucket(userInfo: UserInfo, bucketName: String): Unit = {
     import system.dispatcher
     def bucketDeleteFn(): Unit = {
-      val credential = getBucketCredential(userInfo)
-      val storage = getStorage(credential)
-      val bucket: Bucket = storage.buckets().get(bucketName).execute()
-
-      //Google doesn't let you delete buckets that are full.
-      //You can either remove all the objects manually, or you can set up lifecycle management on the bucket.
-      //This can be used to auto-delete all objects next time the Google lifecycle manager runs (~every 24h).
-      //More info: http://bit.ly/1WCYhhf and http://bit.ly/1Py6b6O
-      val deleteEverythingRule = new Lifecycle.Rule()
-        .setAction(new Action().setType("delete"))
-        .setCondition(new Condition().setAge(0))
-
-      val lifecycle = new Lifecycle().setRule(List(deleteEverythingRule))
-      bucket.setLifecycle(lifecycle)
-
-      Try(storage.buckets().delete(bucketName).execute) match {
+      val buckets = getStorage(getBucketCredential(userInfo)).buckets
+
+      Try(buckets.delete(bucketName).execute) match {
         //Google returns 409 Conflict if the bucket isn't empty.
         case Failure(gjre: GoogleJsonResponseException) if gjre.getDetails.getCode == 409 =>
+          //Google doesn't let you delete buckets that are full.
+          //You can either remove all the objects manually, or you can set up lifecycle management on the bucket.
+          //This can be used to auto-delete all objects next time the Google lifecycle manager runs (~every 24h).
+          //More info: http://bit.ly/1WCYhhf and http://bit.ly/1Py6b6O
+          val deleteEverythingRule = new Lifecycle.Rule()
+            .setAction(new Action().setType("delete"))
+            .setCondition(new Condition().setAge(0))
+          val lifecycle = new Lifecycle().setRule(List(deleteEverythingRule))
+          val bucket = buckets.get(bucketName).execute()
+          bucket.setLifecycle(lifecycle)
+
```

Comment:
It isn't.  The code is reordered for slightly improved sanity:  It isn't necessary to create the lifecycle rule and schedule deletions if you can delete the bucket.  So we try that first.